api/swagger.yml

```diff
@@ -1814,6 +1820,34 @@ paths:
           description: too many requests
         default:
           $ref: "#/components/responses/ServerError"
+
+  /auth/external/login:
+    post:
+      tags:
+        - external
+        - experimental
+      operationId: externalPrincipalLogin
+      summary: perform a login using an external authenticator
+      security: []
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/IdentityToken"
```

Comment:
That request should be an object similar to `StsAuthRequest`, i don't know if lakeFS needs to be aware this field. it should be an envelope, just passing it to the service. 

**Also a duration field needs to be added to the login request that lakeFS does need to know because it generates the JWT itself.**

i.e login with token for 3h.